File: packages/ruby/cognistration/lib/cognistration.rb
require "json"
require "net/http"
require "uri"

module Cognistration
  class Client
    def initialize(base_url = "https://cognistration.com", access_token: nil)
      @base_url = base_url.sub(%r{/\z}, "")
      @access_token = access_token
    end

    def mcp(method, params = {}, request_id: "ruby-sdk-1")
      uri = URI("#{@base_url}/api/mcp")
      request = Net::HTTP::Post.new(uri)
      request["Content-Type"] = "application/json"
      request["MCP-Protocol-Version"] = "2026-07-28"
      request["Mcp-Method"] = method
      request["Authorization"] = "Bearer #{@access_token}" if @access_token
      request.body = JSON.generate(jsonrpc: "2.0", id: request_id, method: method, params: params)
      response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") { |http| http.request(request) }
      JSON.parse(response.body)
    end
  end
end

File: packages/cognistration-cli/bin/cognistration.mjs
#!/usr/bin/env node

const baseUrl = (process.env.COGNISTRATION_URL || 'https://cognistration.com').replace(/\/$/, '');
const command = process.argv[2] || 'capabilities';
const routes = {
  capabilities: '/api/capabilities',
  tools: '/api/mcp',
  docs: '/docs.md',
  sandbox: '/api/sandbox'
};

if (!routes[command]) {
  console.error(`Unknown command: ${command}. Use capabilities, tools, docs, or sandbox.`);
  process.exitCode = 2;
} else {
  const response = command === 'tools'
    ? await fetch(`${baseUrl}${routes[command]}`, { method: 'POST', headers: { 'content-type': 'application/json', 'MCP-Protocol-Version': '2026-07-28', 'Mcp-Method': 'tools/list' }, body: JSON.stringify({ jsonrpc: '2.0', id: 'cli-1', method: 'tools/list', params: {} }) })
    : await fetch(`${baseUrl}${routes[command]}`);
  console.log(JSON.stringify(await response.json(), null, 2));
  if (!response.ok) process.exitCode = 1;
}
